feature/gribi/otg_tests/vrf_policy_driven_te/vrf_policy_driven_te_test.go

```diff
@@ -0,0 +1,1946 @@
+		outHdrSrcIP: ipv4OuterSrc111, outHdrDstIP: traffiDstIP3, InnHdrSrcIP: atePort1.IPv4,
+		InnHdrDstIP: ipv4InnerDst, isInnHdrV4: true, outHdrDscp: []uint32{dscpEncapNoMatch}})
+
+	sendTraffic(t, args, portList, []gosnappi.Flow{flow1, flow2, flow3})
+	verifyTraffic(t, args, []string{"flow1", "flow2", "flow3"}, !wantLoss)
+	captureAndValidatePackets(t, args, &packetValidation{portName: portList[0],
+		outDstIP: []string{traffiDstIP1}, inHdrIP: ipv4InnerDst, validateTTL: false, validateDecap: true})
+
+	t.Run("Send traffic to non decap route and verify the behavior", func(t *testing.T) {
+		flow4 := createFlow(&flowArgs{flowName: flowNegTest, srcPort: atePort1, dstPortList: dstPorts,
+			outHdrSrcIP: ipv4OuterSrc111, outHdrDstIP: ipv4OuterDst333, isInnHdrV4: true,
+			InnHdrSrcIP: atePort1.IPv4, InnHdrDstIP: ipv4InnerDst})
+		sendTraffic(t, args, portList, []gosnappi.Flow{flow4})
+		verifyTraffic(t, args, []string{flowNegTest}, !wantLoss)
+		captureAndValidatePackets(t, args, &packetValidation{portName: portList[0],
+			outDstIP: []string{ipv4OuterDst333}, inHdrIP: ipv4InnerDst, validateNoDecap: true})
+	})
+}
+
+// testGribiDecapMatchSrcProtoDSCP is to validate subtest 2.
+// Test-2, match on source, protocol and DSCP, VRF_DECAP hit -> VRF_ENCAP_A miss -> DEFAULT
+func testGribiDecapMatchSrcProtoDSCP(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice, args *testArgs) {
+	t.Helper()
+	cases := []struct {
+		desc           string
+		prefixWithMask string
+	}{{
+		desc:           "Mask Length 24",
+		prefixWithMask: "192.51.100.0/24",
+	}, {
+		desc:           "Mask Length 32",
+		prefixWithMask: "192.51.100.64/32",
+	}, {
+		desc:           "Mask Length 28",
+		prefixWithMask: "192.51.100.64/28",
+	}, {
+		desc:           "Mask Length 22",
+		prefixWithMask: "192.51.100.0/22",
+	}}
+
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			t.Log("Flush existing gRIBI routes before test.")
+			if err := gribi.FlushAll(args.client); err != nil {
+				t.Fatal(err)
+			}
+
+			// Configure GRIBi baseline AFTs.
+			configGribiBaselineAFT(ctx, t, dut, args)
+
+			t.Run("Program gRIBi route", func(t *testing.T) {
+				configureGribiRoute(ctx, t, dut, args, tc.prefixWithMask)
+			})
+			// Send both 6in4 and 4in4 packets. Verify that the packets have their outer
+			// v4 header stripped and are forwarded according to the route in the DEFAULT
+			// VRF that matches the inner IP address.
+			portList := []string{"port8"}
+			dstPorts := []attrs.Attributes{atePort2, atePort3, atePort4, atePort5, atePort6, atePort7, atePort8}
+			t.Run("Create ip-in-ip and ipv6-in-ip flows, send traffic and verify decap functionality",
+				func(t *testing.T) {
+
+					flow1 := createFlow(&flowArgs{flowName: flow4in4, srcPort: atePort1, dstPortList: dstPorts,
+						outHdrSrcIP: ipv4OuterSrc111, outHdrDstIP: ipv4OuterDst111, outHdrDscp: []uint32{dscpEncapA1},
+						InnHdrSrcIP: atePort1.IPv4, InnHdrDstIP: ipv4InnerDstNoEncap, isInnHdrV4: true})
```

Comment:
Inner IP 138.0.11.12 will match with the 138.0.11.0/24 route in Encap VRF installed with base gRIBI programming ?
Need to use a different inner IP or maybe flush encap vrf gRIBI entries.

Reply:
we are routing here based on pbf and it should take ni based on policy configured.

Reply:
Yes , so here its supposed match on source, protocol and DSCP, VRF_DECAP hit -> VRF_ENCAP_A miss -> DEFAULT ?
But in the ENCAP VRF_A there is already a 138.0.11.0/24 route, so there will be no route match miss right ?